packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/04-upgradeable-plugin/05-updating-versions.md

```diff
@@ -0,0 +1,357 @@
+    if (_number == number) revert AlreadyStored();
+
+    number = _number;
+
+    emit NumberStored({number: _number});
+  }
+
+  function storeAccount(address _account) external auth(STORE_ACCOUNT_PERMISSION_ID) {
+    if (_account == account) revert AlreadyStored();
+
+    account = _account;
+
+    emit AccountStored({account: _account});
+  }
+}
+```
+
+</details>
+
+With each new build implementation, we will need to udate the Plugin Setup contract to be able to update to that new version. We do this through updating the `prepareUpdate()` function with the latest version we've deployed.
+
+```solidity
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity 0.8.17;
+
+import {PermissionLib} from '@aragon/osx/core/permission/PermissionLib.sol';
+import {PluginSetup, IPluginSetup} from '@aragon/osx/framework/plugin/setup/PluginSetup.sol';
+import {SimpleStorageBuild2} from '../build2/SimpleStorageBuild2.sol';
+import {SimpleStorageBuild3} from './SimpleStorageBuild3.sol';
+
+/// @title SimpleStorageSetup build 3
+contract SimpleStorageBuild3Setup is PluginSetup {
+  address private immutable simpleStorageImplementation;
+
+  constructor() {
+    simpleStorageImplementation = address(new SimpleStorageBuild3());
+  }
+
+  /// @inheritdoc IPluginSetup
+  function prepareInstallation(
+    address _dao,
+    bytes memory _data
+  ) external returns (address plugin, PreparedSetupData memory preparedSetupData) {
+    (uint256 _number, address _account) = abi.decode(_data, (uint256, address));
+
+    plugin = createERC1967Proxy(
+      simpleStorageImplementation,
+      abi.encodeWithSelector(SimpleStorageBuild3.initializeBuild3.selector, _dao, _number, _account)
+    );
+
+    PermissionLib.MultiTargetPermission[]
+      memory permissions = new PermissionLib.MultiTargetPermission[](2);
+
+    permissions[0] = PermissionLib.MultiTargetPermission({
+      operation: PermissionLib.Operation.Grant,
+      where: plugin,
+      who: _dao,
+      condition: PermissionLib.NO_CONDITION,
+      permissionId: SimpleStorageBuild3(this.implementation()).STORE_NUMBER_PERMISSION_ID()
+    });
+
+    permissions[1] = permissions[0];
+    permissions[1].permissionId = SimpleStorageBuild3(this.implementation())
+      .STORE_ACCOUNT_PERMISSION_ID();
+
+    preparedSetupData.permissions = permissions;
+  }
+
+  /// @inheritdoc IPluginSetup
+  function prepareUpdate(
+    address _dao,
+    uint16 _currentBuild,
+    SetupPayload calldata _payload
+  )
+    external
+    view
+    override
+    returns (bytes memory initData, PreparedSetupData memory preparedSetupData)
+  {
+    if (_currentBuild == 0) {
+      address _account = abi.decode(_payload.data, (address));
+      initData = abi.encodeWithSelector(
+        SimpleStorageBuild3.initializeFromBuild1.selector,
+        _account
+      );
+    } else if (_currentBuild == 1) {
+      initData = abi.encodeWithSelector(SimpleStorageBuild3.initializeFromBuild2.selector);
```

Comment:
same, `initializeFrom(2)`

Reply:
Simple storage plugin update - added to notes!